### DISCUSSION TITLE
Connecting an array of duplicate ids

### DISCUSSION THREAD

**Discussion by tika** [prisma/prisma#14618] [Q&A] [answered]
Hi!

I'm looking to connect an array of many (potentially) duplicate item ids. Right now the code below allows for UNIQUE IDS, however I would specifically like to add many of the same item.

```
       const order = await prisma.order.create({
        data: {
                note: data.note ?? "",
                items: {
                    connect: arr,
                },
                userId: data.userId,
                restaurantId: data.restaurantId,
            },
        });
```

Thanks,
**tika.**

**Comment by tika** [comment]
Do let me know of any potential solutions, if none arise, I will redesign my Item model.
**tika**.

**Comment by alii** [comment, marked as answer]
> Edit: as referenced in the replies to the thread, the accepted answer was:
> > Perhaps it would be wise to then have another model called OrderItem that has a .itemId and a .quantity?

Perhaps you should deuplicate the array in JavaScript first hand. This sounds like an issue outside of scope for Prisma.

For example, if `arr` is an array of primitives, you could use `Set` and dedupde as follows
```ts
declare const arr: string[];

const dedupedArray = [...new Set(arr)];

const order = await prisma.order.create({
  data: {
    note: data.note ?? "",
    items: {
      connect: dedupedArray,
    },
    userId: data.userId,
    restaurantId: data.restaurantId,
  },
});

```

**Reply by tika** [reply]
Thanks for the response!

However, I would like to be able to have duplicate items - I don't think prisma allows for this behaviour but I'm just wondering if it's possible.

So here is the data that I'm passing into the creation items:
```
[
  { id: 5 }, { id: 5 },
  { id: 5 }, { id: 5 },
  { id: 5 }, { id: 5 },
  { id: 5 }, { id: 4 },
  { id: 4 }, { id: 8 },
  { id: 8 }, { id: 8 },
  { id: 3 }, { id: 3 }
]
```

And when querying the order (after creation), this is the response of order.items:
```
[
  {
    id: 3,
    image: '',
    name: 'Test',
    price: 5,
    description: '',
    category: 'curries',
    restaurantId: 1
  },
  {
    id: 4,
    image: 'https://res.cloudinary.com/demo/image/fetch/https://hips.hearstapps.com/del.h-cdn.co/assets/17/31/1501791674-delish-chicken-curry-horizontal.jpg?crop=1.00xw:0.750xh;0,0.159xh&resize=1200:*',
    name: 'Madras',
    price: 5,
    description: '',
    category: 'curries',
    restaurantId: 1
  },
  {
    id: 5,
    image: '',
    name: 'Pancakes',
    price: 5,
    description: '',
    category: 'curries',
    restaurantId: 1
  },
  {
    id: 8,
    image: '',
    name: 'test',
    price: 5,
    description: '',
    category: 'curries',
    restaurantId: 1
  }
]
```

Notice how it's removed all the duplicates, which is not what I want in this case. Alternatively, I can add an "amount" field to the item, which would allow me to pass in the amount I'm wanting, however, do let me know if there's any better solutions to that.

**Reply by alii** [reply]
Perhaps it would be wise to then have another model called OrderItem that has a .itemId and a .quantity?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
